todo_app/data/ViewModel.py

```diff
@@ -4,63 +4,65 @@
-    self._todolist = todolist
-    self._doinglist = doinglist
-    self._donelist = donelist    
-  
-  @property
-  def todolist(self):
-     return self._todolist
-     
-  @property
-  def doinglist(self):
-     return self._doinglist
-  
-  @property
-  def donelist(self):
-     return self._donelist
- 
+    def __init__(self, todolist, doinglist, donelist):
+
+        self._todolist = todolist
+        self._doinglist = doinglist
+        self._donelist = donelist
+
+    @property
+    def todolist(self):
+        return self._todolist
+
+    @property
+    def doinglist(self):
+        return self._doinglist
+
+    @property
+    def donelist(self):
+        return self._donelist
+
+    def show_all_done_items(self):
+        today = date.today()
+
+        number = len(self.donelist)
+        if number < 5:
+            return True
+        else:
+            return False
 
-  
+    def recent_done_items(self):
+        dbconnect = os.environ['CLIENT']
+        now = datetime.now()
+        date_time = now.strftime("%Y-%m-%d")
+        client = pymongo.MongoClient(
+            dbconnect, ssl=True, ssl_cert_reqs='CERT_NONE')
 
-  def show_all_done_items(self):
-    today = date.today()
-    
-    number = len(self.donelist)
-    if number < 5:
-      return True
-    else: 
-      return False
-   
+        database = client["01"]
+        trello_collection = database["trello_collection"]
+        date_now = datetime.now().strftime('%Y-%m-%d')
+        result = trello_collection.find(
+            {"idBoard": "6005828032dafa5707bf5dc5",
+             "last_modified": date_now}
+        )
 
-  def recent_done_items(self):
-    
-    now = datetime.now()
-    date_time = now.strftime("%Y-%m-%d")
-    
-    
-    
-    return [item for item in self.donelist if datetime.strptime(item.dateLastActivity,"%Y-%m-%dT%H:%M:%S.%fZ").date() == now.date() ]
-    
-    
+        return [item for item in self.donelist if datetime.strptime(item.last_modified, '%Y-%m-%d').date() == now.date()]
 
-  def older_done_items(self):
-   
-    now = datetime.now()
-    date_time = now.strftime("%Y-%m-%d")
-	  
-    
-    
-    return [item for item in self.donelist if datetime.strptime(item.dateLastActivity,"%Y-%m-%dT%H:%M:%S.%fZ").date() != now.date() ]
-    
-  
+    def older_done_items(self):
 
+        now = datetime.now()
+        date_time = now.strftime("%Y-%m-%d")
 
-  
+        client = pymongo.MongoClient(
+            "mongodb+srv://admin:MongoAdmin1@cluster0.qtpde.mongodb.net/myFirstDatabase?retryWrites=true&w=majority", ssl=True, ssl_cert_reqs='CERT_NONE')
+        database = client["01"]
+        trello_collection = database["trello_collection"]
+        date_now = datetime.now().strftime('%Y-%m-%d')
+        result = trello_collection.find(
+            {"idBoard": "6005828032dafa5707bf5dc5", "last_modified": date_now})
 
+        return [item for item in self.donelist if datetime.strptime(item.last_modified, "%Y-%m-%d").date() != now.date()]
```

Comment:
If you look carefully at what these functions are doing, they actually do quite a lot of work that doesn't get used at all!

For example, for the first one, we:
* Connect to the database
* Find all the matching items
* Then ignore them (!), and just compare the dates for our done items

So we could reduce this significantly down to:
```suggestion
    def recent_done_items(self):
        now = datetime.now()
        return [item for item in self.donelist if datetime.strptime(item.last_modified, '%Y-%m-%d').date() == now.date()]

    def older_done_items(self):

        now = datetime.now()
        return [item for item in self.donelist if datetime.strptime(item.last_modified, "%Y-%m-%d").date() != now.date()]

```

This would then fix the issue with your items_test currently failing on Travis - we could also fix that by providing the relevant environment variable but in our case there's no harm in just removing the need for it in the first place

Reply:
Updated, thank you